<?xml version="1.0"?>
  <robot name="move_vehicle">

<!--  Define the baseprint of the robot  -->
  <link name="base_footprint">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0" />
      <geometry>
        <box size="0.05 0.05 0.001" />
      </geometry>
      <material name="MistyRose">
        <color  rgba="1 0.8945 0.8828 1" />
      </material>
    </visual>
  </link>

<!--  Define the robot movebase  -->
    <link name="base_link">
	<inertial>
        <origin xyz="0.038037 -0.0019933 0.00077981" rpy="0 0 0" />
        <mass value="31.065" />
        <inertia ixx="1.0666" ixy="0.011171" ixz="-0.00016117"
                 iyy="2.3226" iyz="0.00014034" izz="3.1096" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/base_link.STL" />
        </geometry>
        <material name="MistyRose">
          <color  rgba="1 0.8945 0.8828 1" />
        </material>
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/base_link.STL" />
        </geometry>
      </collision>
    </link>
    <joint name="base_joint" type="fixed">
      <origin rpy="0 0 0" xyz="0 0 0.175"/>
      <parent link="base_footprint"/>
      <child link="base_link"/>
    </joint>

<!--  Define the fake joints for the move base  -->
    <link name="fake_left_wheel">
      <visual>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
          <box size="0.001 0.001 0.001"/>
        </geometry>
      </visual>
      <collision>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
          <box size="0.001 0.001 0.001"/>
        </geometry>
      </collision>
    </link>

    <joint name="base_l_wheel_joint" type="continuous">
      <origin xyz="0.0 0.3901 0.0" rpy="-1.5708 -1.5708 0" />
      <parent link="base_link" />
      <child link="fake_left_wheel" />
      <axis xyz="0 0 1" />
    </joint>

    <link name="fake_right_wheel">
      <visual>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
          <box size="0.001 0.001 0.001"/>
        </geometry>
      </visual>
      <collision>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
          <box size="0.001 0.001 0.001"/>
        </geometry>
      </collision>
    </link>

    <joint name="base_r_wheel_joint" type="continuous">
      <origin xyz="0.0 -0.3901 0.0" rpy="-1.5708 -1.5708 0" />
      <parent link="base_link" />
      <child link="fake_right_wheel" />
      <axis xyz="0 0 1" />
    </joint>

<!--  Define the four wheels of robot movebase  -->
    <link name="wheel_lf">
      <inertial>
        <origin xyz="1.7209E-06 -7.4175E-07 -0.024764" rpy="0 0 0" />
        <mass value="7.6485" />
        <inertia ixx="0.079312" ixy="9.0434E-07" ixz="-5.3408E-07"
                 iyy="0.079311" iyz="2.374E-07" izz="0.12138" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/wheel_lf.STL" />
        </geometry>
        <material name="Grey">
          <color rgba="0.2148 0.2148 0.2148 1" />
        </material>
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/wheel_lf.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_wheel_lf" type="continuous">
      <origin xyz="0.319 0.3901 0.0" rpy="-1.5708 -1.5708 0" />
      <parent link="base_link" />
      <child link="wheel_lf" />
      <axis xyz="0 0 1" />
      <limit effort="1000" velocity="0.7854" />
      <mimic joint="base_l_wheel_joint" multiplier="1" offset="0" />
    </joint>

    <!-- joint name="joint_wheel_lf" type="fixed">
      <origin xyz="0.319 0.3901 0.0" rpy="-1.5708 -1.5708 0" />
      <parent link="base_link" />
      <child link="wheel_lf" />
    </joint -->
  
    <link name="wheel_rf">
      <inertial>
        <origin xyz="-1.8953E-06 -2.7769E-08 0.024816" rpy="0 0 0" />
        <mass value="7.553" />
        <inertia ixx="0.079028" ixy="2.7999E-08" ixz="-5.8191E-07"
                 iyy="0.079026" iyz="-1.5482E-08" izz="0.12098" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/wheel_rf.STL" />
        </geometry>
        <material name="Grey" />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/wheel_rf.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_wheel_rf" type="continuous">
      <origin xyz="0.319 -0.3901 0.0" rpy="-1.5708 -1.5708 0" />
      <parent link="base_link" />
      <child link="wheel_rf" />
      <axis xyz="0 0 1" />
      <limit effort="1000" velocity="0.7854" />
      <mimic joint="base_r_wheel_joint" multiplier="1" offset="0" />
    </joint>

    <!-- joint name="joint_wheel_rf" type="fixed">
      <origin xyz="0.319 -0.3901 0.0" rpy="-1.5708 -1.5708 0" />
      <parent link="base_link" />
      <child link="wheel_rf" />
    </joint -->
  
    <link name="wheel_lb">
      <inertial>
        <origin xyz="-1.5753E-06 1.2458E-06 -0.014717" rpy="0 0 0" />
        <mass value="7.133" />
        <inertia ixx="0.065774" ixy="1.2549E-06" ixz="5.6643E-07"
                 iyy="0.065773" iyz="-4.5681E-07" izz="0.11221" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/wheel_lb.STL" />
        </geometry>
        <material name="Grey" />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/wheel_lb.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_wheel_lb" type="continuous">
      <origin xyz="-0.319 0.3901 0.0" rpy="-1.5708 -1.5708 0" />
      <parent link="base_link" />
      <child link="wheel_lb" />
      <axis xyz="0 0 1" />
      <limit effort="1000" velocity="0.7854" />
      <mimic joint="base_l_wheel_joint" multiplier="1" offset="0" />
    </joint>

    <!-- joint name="joint_wheel_lb" type="fixed">
      <origin xyz="-0.319 0.3901 0.0" rpy="-1.5708 -1.5708 0" />
      <parent link="base_link" />
      <child link="wheel_lb" />
    </joint -->
  
    <link name="wheel_rb">
      <inertial>
        <origin xyz="-1.259E-06 1.5919E-06 0.014292" rpy="0 0 0" />
        <mass value="7.0585" />
        <inertia ixx="0.065491" ixy="1.2549E-06" ixz="-4.6059E-07"
                 iyy="0.065492" iyz="5.712E-07" izz="0.1119" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/wheel_rb.STL" />
        </geometry>
        <material name="Grey" />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/wheel_rb.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_wheel_rb" type="continuous">
      <origin xyz="-0.319 -0.3901 0.0" rpy="-1.5708 -1.5708 0" />
      <parent link="base_link" />
      <child link="wheel_rb" />
      <axis xyz="0 0 1" />
      <limit effort="1000" velocity="0.7854" />
      <mimic joint="base_r_wheel_joint" multiplier="1" offset="0" />
    </joint>

    <!-- joint name="joint_wheel_rb" type="fixed">
      <origin xyz="-0.319 -0.3901 0.0" rpy="-1.5708 -1.5708 0" />
      <parent link="base_link" />
      <child link="wheel_rb" />
    </joint -->

<!--  Define the left swing-arm and its follow-up wheel  -->
    <link name="swingarm_left">
      <inertial>
        <origin xyz="-0.0040312 0.13583 -0.011794" rpy="0 0 0" />
        <mass value="4.1371" />
        <inertia ixx="0.14405" ixy="-0.002008" ixz="-0.00056728"
                 iyy="0.025525" iyz="-0.00019287" izz="0.1556" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/swingarm_left.STL" />
        </geometry>
        <material name="LightGrey">
          <color rgba="0.75 0.75 0.75 1" />
        </material>
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/swingarm_left.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_swingarm_left" type="revolute">
      <origin xyz="0.319 0.2516 0.0" rpy="-1.5708 -1.5708 0" />
      <parent link="base_link" />
      <child link="swingarm_left" />
      <axis xyz="0 0 -1" />
      <limit lower="-0.7854" upper="1.0472" effort="1000" velocity="0.7854" />
    </joint>

    <!-- joint name="joint_swingarm_left" type="fixed">
      <origin xyz="0.319 0.2516 0.0" rpy="-1.5708 -1.5708 0" />
      <parent link="base_link" />
      <child link="swingarm_left" />
    </joint -->

<!--  Define the right swing-arm and its follow-up wheel  -->
    <link name="swingarm_right">
      <inertial>
        <origin xyz="-0.0040022 0.13616 0.010996" rpy="0 0 0" />
        <mass value="4.1247" />
        <inertia ixx="0.14347" ixy="-0.0020306" ixz="0.00055968"
                 iyy="0.025147" iyz="0.00063656" izz="0.15536" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/swingarm_right.STL" />
        </geometry>
        <material name="LightGrey" />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/swingarm_right.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_swingarm_right" type="revolute">
      <origin xyz="0.319 -0.2516 0.0" rpy="-1.5708 -1.5708 0" />
      <parent link="base_link" />
      <child link="swingarm_right" />
      <axis xyz="0 0 -1" />
      <limit lower="-0.7854" upper="1.0472" effort="1000" velocity="0.7854" />
    </joint>

    <!-- joint name="joint_swingarm_right" type="fixed">
      <origin xyz="0.319 -0.2516 0.0" rpy="-1.5708 -1.5708 0" />
      <parent link="base_link" />
      <child link="swingarm_right" />
    </joint -->

<!--  Define the base of the 7DOF arm  -->
    <link name="11_arm_base_link">
      <inertial>
        <origin xyz="0.0005196 -0.002636 0.02178" rpy="0 0 0" />
        <mass value="0.5648" />
        <inertia ixx="0.001421" ixy="-2.8620E-06" ixz="-4.8464E-06"
                 iyy="0.001546" iyz="2.3241E-05" izz="0.002700" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/11_arm_base_link.STL" />
        </geometry>
        <material name="Bisque">
          <color rgba="0.9336 0.9336 0.7188 1" />
        </material>
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/11_arm_base_link.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_arm_base_link" type="fixed">
      <!--origin xyz="0.2834 0.0506 0.2895" rpy="0 0 0" /-->
      <origin xyz="0.2834 0.0506 0.1145" rpy="0 0 0" />
      <parent link="base_link" />
      <child link="11_arm_base_link" />
      <axis xyz="0 0 0" />
    </joint>

<!--  Define the turret-ratate part of the 7DOF arm  -->
    <link name="12_arm_turret_rotate">
      <inertial>
        <origin xyz="0.017985 -0.0088737 0.11066" rpy="0 0 0" />
        <mass value="4.3003" />
        <inertia ixx="0.033193" ixy="-0.0035111" ixz="0.0011378"
                 iyy="0.032315" iyz="-0.0011051" izz="0.036273" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/12_arm_turret_rotate.STL" />
        </geometry>
        <material name="Bisque" />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/12_arm_turret_rotate.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_turret_rotate" type="revolute">
      <origin xyz="0 0 0.06" rpy="0 0 0" />
      <parent link="11_arm_base_link" />
      <child link="12_arm_turret_rotate" />
      <axis xyz="0 0 1" />
      <limit lower="-3.1415" upper="3.1415" effort="1000" velocity="0.7854" />
    </joint>

<!--  Define the bigarm-pitch part of the 7DOF arm  -->
    <link name="13_arm_bigarm_pitch">
      <inertial>
        <origin xyz="0.10588 0.0046944 0.046951" rpy="0 0 0" />
        <mass value="1.87495" />
        <inertia ixx="0.0053558" ixy="-0.00049012" ixz="0.0014335"
                 iyy="0.036157" iyz="5.0898E-05" izz="0.037051" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/13_arm_bigarm_pitch.STL" />
        </geometry>
        <material name="Bisque" />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/13_arm_bigarm_pitch.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_bigarm_pitch" type="revolute">
      <origin xyz="0.060569 -0.1185 0.1095" rpy="1.5708 -1.5708 0" />
      <parent link="12_arm_turret_rotate" />
      <child link="13_arm_bigarm_pitch" />
      <axis xyz="0 0 1" />
      <limit lower="-1.9199" upper="1.1345" effort="1000" velocity="0.7854" />
    </joint>

<!--  Define the bigarm-linear part of the 7DOF arm  -->
    <link name="14_arm_bigarm_linear">
      <inertial>
        <origin xyz="0.016875 -0.0058662 -0.030993" rpy="0 0 0" />
        <mass value="1.1107" />
        <inertia ixx="0.018966" ixy="-7.4160E-05" ixz="0.0018801"
                 iyy="0.019599" iyz="-0.00058459" izz="0.0019633" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/14_arm_bigarm_linear.STL" />
        </geometry>
        <material name="Bisque" />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/14_arm_bigarm_linear.STL" />
        </geometry>
      </collision>
    </link>

    <!-- joint name="joint_bigarm_linear" type="prismatic">
      <origin xyz="0.426 0 0.054" rpy="0 1.5708 0" />
      <parent link="13_arm_bigarm_pitch" />
      <child link="14_arm_bigarm_linear" />
      <axis xyz="0 0 1" />
      <limit lower="0" upper="0.2" effort="1000" velocity="0.2" />
    </joint -->

    <joint name="joint_bigarm_linear" type="fixed">
      <origin xyz="0.486 0 0.054" rpy="0 1.5708 0" />
      <parent link="13_arm_bigarm_pitch" />
      <child link="14_arm_bigarm_linear" />
    </joint>

<!--  Define the forearm-pitch part of the 7DOF arm  -->
    <link name="15_arm_forearm_pitch">
      <inertial>
        <origin xyz="0.065889 -0.0025453 -0.015685" rpy="0 0 0" />
        <mass value="0.82182" />
        <inertia ixx="0.0012808" ixy="8.8555E-05" ixz="0.00016596"
                 iyy="0.0035941" iyz="7.0432E-05" izz="0.0035121" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/15_arm_forearm_pitch.STL" />
        </geometry>
        <material name="Bisque" />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/15_arm_forearm_pitch.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_forearm_pitch" type="revolute">
      <origin xyz="0.08818 -0.01 0.075743" rpy="1.5708 0 -1.5708" />
      <parent link="14_arm_bigarm_linear" />
      <child link="15_arm_forearm_pitch" />
      <axis xyz="0 0 -1" />
      <limit lower="-2.7053" upper="2.7053" effort="1000" velocity="0.7854" />
    </joint>

<!--  Define the forearm-rotate part of the 7DOF arm  -->
    <link name="16_arm_forearm_rotate">
      <inertial>
        <origin xyz="0.022732 0.00015966 0.26742" rpy="0 0 0" />
        <mass value="1.47474" />
        <inertia ixx="0.028919" ixy="1.6692E-06" ixz="0.0043549"
                 iyy="0.030231" iyz="1.8894E-05" izz="0.0030219" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/16_arm_forearm_rotate.STL" />
        </geometry>
        <material name="Bisque" />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/16_arm_forearm_rotate.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_forearm_rotate" type="revolute">
      <origin xyz="0.1585 0 -0.01" rpy="0 1.5708 0" />
      <parent link="15_arm_forearm_pitch" />
      <child link="16_arm_forearm_rotate" />
      <axis xyz="0 0 -1" />
      <limit lower="-3.1415" upper="3.1415" effort="1000" velocity="0.7854" />
    </joint>

<!--  Define the wrist-pitch part of the 7DOF arm  -->
    <link name="17_arm_wrist_pitch">
      <inertial>
        <origin xyz="0.093903 -8.3831E-05 0.025644" rpy="0 0 0" />
        <mass value="0.87731" />
        <inertia ixx="0.0012299" ixy="-1.7286E-05" ixz="0.0001045"
                 iyy="0.0048959" iyz="-1.0051E-05" izz="0.0049504" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/17_arm_wrist_pitch.STL" />
        </geometry>
        <material name="Bisque" />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/17_arm_wrist_pitch.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_wrist_pitch" type="revolute">
      <origin xyz="0.026285 0 0.45399" rpy="0 -1.5708 0" />
      <parent link="16_arm_forearm_rotate" />
      <child link="17_arm_wrist_pitch" />
      <axis xyz="0 0 1" />
      <limit lower="-1.5708" upper="1.5708" effort="1000" velocity="0.7854" />
    </joint>

<!--  Define the wrist-rotate part of the 7DOF arm  -->
    <link name="18_arm_wrist_rotate">
      <inertial>
        <origin xyz="7.8716E-05 -9.5743E-06 0.034827" rpy="0 0 0" />
        <mass value="0.18321" />
        <inertia ixx="0.00012884" ixy="2.9321E-07" ixz="2.5543E-07"
                 iyy="0.00025515" iyz="-2.0848E-08" izz="0.00023184" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/18_arm_wrist_rotate.STL" />
        </geometry>
        <material name="Bisque" />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/18_arm_wrist_rotate.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_wrist_rotate" type="revolute">
      <origin xyz="0.2125 0 0.0155" rpy="0 1.5708 0" />
      <parent link="17_arm_wrist_pitch" />
      <child link="18_arm_wrist_rotate" />
      <axis xyz="0 0 -1" />
      <limit lower="-3.1415" upper="3.1415" effort="1000" velocity="0.7854" />
    </joint>

<!--  Fake gripper link and joint for grasp planning  -->
  <link name="fake_arm_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.001 0.001 0.001"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>

  <joint name="fake_arm_joint" type="fixed">
    <origin rpy="0 0 1.5708" xyz="0 0 0.2179"/>
    <parent link="18_arm_wrist_rotate"/>
    <child link="fake_arm_link"/>
  </joint>

<!--  Define the left gripper-rod part1 of the gripper  -->
    <link name="gripper_rod_left1">
      <inertial>
        <origin xyz="0.0392189 0.0017462 -0.00013931" rpy="0 0 0" />
        <mass value="0.024983" />
        <inertia ixx="6.0516E-06" ixy="1.0687E-07" ixz="-2.3690E-07"
                 iyy="1.8505E-05" iyz="-9.6717E-09" izz="1.3485E-05" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/gripper_rod_left1.STL" />
        </geometry>
        <material name="Bisque" />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/gripper_rod_left1.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_rod_left1" type="revolute">
      <origin xyz="0.052 0 0.054" rpy="-1.5708 -0.4276 0" /> 
      <parent link="18_arm_wrist_rotate" />
      <child link="gripper_rod_left1" />
      <axis xyz="0 0 -1" />
      <limit lower="0" upper="1.2758" effort="1000" velocity="0.7854"/>
    </joint>

<!--  Define the left gripper tip of the gripper  -->
    <link name="gripper_tip_left">
      <inertial>
        <origin xyz="0.070142 -0.029074 -0.00048704" rpy="0 0 0" />
        <mass value="0.085742" />
        <inertia ixx="1.3572E-05" ixy="-1.4399E-05" ixz="1.3409E-08"
                 iyy="0.00019024" iyz="6.2787E-09" izz="0.00019252" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/gripper_tip_left.STL" />
        </geometry>
        <material name="Bisque" />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/gripper_tip_left.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_tip_left" type="revolute">
      <origin xyz="0.072 0 0" rpy="0 0 -1.1432" />
      <parent link="gripper_rod_left1" />
      <child link="gripper_tip_left" />
      <axis xyz="0 0 1" />
      <limit lower="0" upper="1.2758" effort="1000" velocity="0.7854" />
      <mimic joint="joint_rod_left1" multiplier="1" offset="0" />
    </joint>

<!--  Define the right gripper-rod part1 of the gripper  -->
    <link name="gripper_rod_right1">
      <inertial>
        <origin xyz="0.039219 -0.0017462 0.00013931" rpy="0 0 0" />
        <mass value="0.024983" />
        <inertia ixx="6.0516E-06" ixy="-1.0687E-07" ixz="2.3690E-07"
                 iyy="1.8505E-05" iyz="-9.6716E-09" izz="1.3485E-05" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/gripper_rod_right1.STL" />
        </geometry>
        <material name="Bisque" />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/gripper_rod_right1.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_rod_right1" type="revolute">
      <origin xyz="-0.052 0 0.054" rpy="-1.5708 -2.7140 0" />
      <parent link="18_arm_wrist_rotate" />
      <child link="gripper_rod_right1" />
      <axis xyz="0 0 1" />
      <limit lower="0" upper="1.2758" effort="1000" velocity="0.7854" />
      <mimic joint="joint_rod_left1" multiplier="1" offset="0" />
    </joint>

<!--  Define the right gripper tip of the gripper  -->
    <link name="gripper_tip_right">
      <inertial>
        <origin xyz="0.070142 0.029074 -0.00051296" rpy="0 0 0" />
        <mass value="0.085742" />
        <inertia ixx="1.3572E-05" ixy="1.4399E-05" ixz="-1.3407E-08"
                 iyy="0.00019024" iyz="6.2766E-09" izz="0.00019252" />
      </inertial>
      <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/gripper_tip_right.STL" />
        </geometry>
        <material name="Bisque" />
      </visual>
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
          <mesh filename="package://mobot6_interface/meshes/gripper_tip_right.STL" />
        </geometry>
      </collision>
    </link>

    <joint name="joint_tip_right" type="revolute">
      <origin xyz="0.072 0 0" rpy="0 0 1.1432" />
      <parent link="gripper_rod_right1" />
      <child link="gripper_tip_right" />
      <axis xyz="0 0 -1" />
      <limit lower="0" upper="1.2758" effort="1000" velocity="0.7854" />
      <mimic joint="joint_rod_left1" multiplier="1" offset="0" />
    </joint>

  </robot>






